frontend/model/contracts/chatroom.js

```diff
@@ -551,6 +568,7 @@ sbp('chelonia/defineContract', {
         }
 
         if (me === innerSigningContractID) {
+          deleteEncryptedFilesByToken(data.manifestCids, innerSigningContractID)
```

Comment:
I'm commenting here, although the 'error' will happen at the action in the identity contract. This logic isn't necessarily correct, because the person sending the message isn't necessarily the one that uploaded the files (see #1881). So, although this particular line here is _probably_ fine from what I can see (other than being in the `process` function), throwing/logging an error when a token isn't found isn't ideal. Also, remember that even if the assumption of `uploader = owner` holds, `process` (and `sideEffect`) can be called multiple times.

Reply:
@corrideat:

> other than being in the `process` function

This is in the `sideEffect` AFAICT...

> `process` (and `sideEffect`) can be called multiple times.

Can you elaborate on how `sideEffect` can be called multiple times? Do you mean because every user in the chatroom will call it at least once? (This is gated by the `if` check...)

@Silver-IT:

Note that per my [other comment](https://github.com/okTurtles/group-income/pull/1892#discussion_r1557878667), `'gi.actions/identity/removeFiles'` will need to throw an exception upon an error. So that means that you either need a `try/catch` here, or inside of `deleteEncryptedFilesByToken` in order to catch and either log or display the error to the user (perhaps via a notification).

Reply:
>> other than being in the process function

Please disregard that, I updated my comment but not that part

> Can you elaborate on how sideEffect can be called multiple times?

`sideEffect` can be called any number of times, for every time a contract gets re-synced from scratch (this goes for `process` and `validate` too, which can be called even more times, in part because of outgoing messages). Hence, expecting it to be called once and reporting errors on successive runs is something to avoid, unless the error is severe enough to merit a (likely) false positive.

Reply:
> Hence, expecting it to be called once and reporting errors on successive runs is something to avoid, unless the error is severe enough to merit a (likely) false positive.

Good point 👍 

So the error can be just logged (not shown to the user), but it still needs to be caught either here or inside of `deleteEncryptedFilesByToken`